lib/shoryuken/manager.rb

```diff
@@ -19,8 +19,11 @@ def initialize(fetcher, polling_strategy)
       @polling_strategy = polling_strategy
 
       @heartbeat = Concurrent::TimerTask.new(run_now: true,
-                                             execution_interval: HEARTBEAT_INTERVAL,
-                                             timeout_interval: 60) { dispatch }
+                                             execution_interval: HEARTBEAT_INTERVAL) { @pool.post { dispatch } if @dispatching.false? }
```

Comment:
Looks good. 

As long as the block finishes faster than the `execution_timeout` then this leak won't occur, even if the currently broken state of `concurrent-ruby`. 

Now if only the team there could be convinced that there is a bug in the first place. 😜